flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryConfiguration.java

```diff
@@ -170,12 +173,39 @@ public static MetricRegistryConfiguration fromConfiguration(Configuration config
 			}
 		}
 
+		List<ReporterSetup> reporterArguments = new ArrayList<>(reporterConfigurations.size());
+		for (Tuple2<String, Configuration> reporterConfiguration: reporterConfigurations) {
+			String reporterName = reporterConfiguration.f0;
+			Configuration reporterConfig = reporterConfiguration.f1;
+
+			try {
+				final String reporterClassName = reporterConfig.getString(ConfigConstants.METRICS_REPORTER_CLASS_SUFFIX, null);
+				if (reporterClassName == null) {
+					LOG.error("No reporter class set for reporter " + reporterName + ". Metrics might not be exposed/reported.");
+					continue;
+				}
+
+				final SupplierWithException<MetricReporter, Exception> supplier = () -> {
+					Class<?> reporterClass = Class.forName(reporterClassName);
+					return (MetricReporter) reporterClass.newInstance();
+				};
+
+				MetricConfig metricConfig = new MetricConfig();
+				reporterConfig.addAllToProperties(metricConfig);
```

Comment:
We could factor this logic out into a separate method e.g. `ReporterSetup#fromConfiguration()` or so.